src/directive/service/get_by_archive_id.test.ts

```diff
@@ -0,0 +1,43 @@
+import { db } from "../../database";
+import { directiveService } from "./index";
+
+jest.mock("../../database");
+
+const testDirectiveId = "39b2a5fa-3508-4030-91b6-21dc6ec7a1ab";
+const testArchiveId = 1;
+
+const loadFixtures = async (): Promise<void> => {
+  await db.sql("fixtures.create_test_accounts");
+  await db.sql("fixtures.create_test_archive");
+  await db.sql("fixtures.create_test_directive");
+  await db.sql("fixtures.create_test_directive_trigger");
+};
+
+const clearDatabase = async (): Promise<void> => {
+  await db.query(
+    "TRUNCATE account, archive, directive, directive_trigger CASCADE"
+  );
+};
+
+describe("getDirectivesByArchiveId", () => {
+  beforeEach(async () => {
+    await clearDatabase();
+    await loadFixtures();
+  });
+  afterEach(async () => {
+    await clearDatabase();
+  });
+
+  test("should return directive from the database", async () => {
+    const response = await directiveService.getDirectivesByArchiveId(
+      testArchiveId
+    );
+    expect(response.length).toBe(1);
+    expect(response[0]?.directiveId).toBe(testDirectiveId);
+  });
+
+  test("should return empty list if archive has no directives", async () => {
+    const response = await directiveService.getDirectivesByArchiveId(9999);
+    expect(response.length).toBe(0);
+  });
+});
```

Comment:
Probably want a "should return empty list if authenticated account does not own archive," assuming that we don't want to confirm a directive exists (i.e., essentially return a 404 rather than a 401)